pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -239,15 +239,61 @@ type reconciledContainer struct {
 	containerID   string
 }
 
-func (m *manager) reconcileState() (success []reconciledContainer, failure []reconciledContainer) {
+func (m *manager) removeStaleState() {
 	if !m.sourcesReady.AllReady() {
 		return
 	}
+
+	m.Lock()
+	defer m.Unlock()
+
+	activePods := m.activePods()
+	if len(activePods) == 0 {
+		// If there are no active pods, then we skip the removeal of stale
+		// state altogether. This ensures we don't remove something
+		// inadvertently and we will just try again next time this function is
+		// called.
+		return
+	}
+
+	activeContainers := make(map[string]struct{})
```

Comment:
just wondering, why are we mapping a string to a struct here instead of something simple like a bool?

Reply:
It's more space efficient since i don't need the value, just the key:
https://medium.com/@l.peppoloni/how-to-improve-your-go-code-with-empty-structs-3bd0c66bc531